src/ucs/sys/sys.c

```diff
@@ -318,20 +318,9 @@ uint64_t ucs_generate_uuid(uint64_t seed)
 
 int ucs_sys_max_open_files()
 {
-    static int file_limit = 0;
     struct rlimit rlim;
-    int ret;
-
-    if (file_limit == 0) {
-        ret = getrlimit(RLIMIT_NOFILE, &rlim);
-        if (ret == 0) {
-            file_limit = (int)rlim.rlim_cur;
-        } else {
-            file_limit = 1024;
-        }
-    }
 
-    return file_limit;
+    return getrlimit(RLIMIT_NOFILE, &rlim) != 0 ? -1 : (int)rlim.rlim_cur;
```

Comment:
can we share commeon code with ucs_sys_get_effective_memlock_rlimit?

Reply:
fixed